libs/sdk-ui-pivot/src/impl/tableFacade.ts

```diff
@@ -0,0 +1,570 @@
+import { TableDescriptor } from "./structure/tableDescriptor";
+import { IDataView, IExecutionResult, IPreparedExecution } from "@gooddata/sdk-backend-spi";
+import {
+    createExportFunction,
+    DataViewFacade,
+    IAvailableDrillTargets,
+    IExportFunction,
+} from "@gooddata/sdk-ui";
+import {
+    AUTO_SIZED_MAX_WIDTH,
+    autoresizeAllColumns,
+    getAutoResizedColumns,
+    isColumnAutoResized,
+    MANUALLY_SIZED_MAX_WIDTH,
+    resetColumnsWidthToDefault,
+    resizeAllMeasuresColumns,
+    ResizedColumnsStore,
+    resizeWeakMeasureColumns,
+    syncSuppressSizeToFitOnColumns,
+    updateColumnDefinitionsWithWidths,
+} from "./resizing/columnSizing";
+import { IResizedColumns, UIClick } from "../columnWidths";
+import { AgGridDatasource, createAgGridDatasource } from "./data/dataSource";
+import { Column, ColumnApi, GridApi } from "@ag-grid-community/all-modules";
+import { defFingerprint, ISortItem } from "@gooddata/sdk-model";
+import { invariant } from "ts-invariant";
+import { IntlShape } from "react-intl";
+import { fixEmptyHeaderItems } from "@gooddata/sdk-ui-vis-commons";
+import { setColumnMaxWidth, setColumnMaxWidthIf } from "./base/agColumnWrapper";
+import { agColIds, isMeasureColumn } from "./base/agUtils";
+import { agColId } from "./structure/tableDescriptorTypes";
+import { sleep } from "./utils";
+import { DEFAULT_AUTOSIZE_PADDING, DEFAULT_ROW_HEIGHT } from "./base/constants";
+import { getAvailableDrillTargets } from "./drilling/drillTargets";
+import { IGroupingProvider } from "./data/rowGroupingProvider";
+import sumBy from "lodash/sumBy";
+import ApiWrapper from "./base/agApiWrapper";
+import {
+    initializeStickyRow,
+    stickyRowExists,
+    updateStickyRowContentClassesAndData,
+    updateStickyRowPosition,
+} from "./stickyRowHandler";
+import { ColumnResizingConfig, TableConfig, StickyRowConfig } from "./privateTypes";
+import { ICorePivotTableProps } from "../publicTypes";
+
+const HEADER_CELL_BORDER = 1;
+const COLUMN_RESIZE_TIMEOUT = 300;
+
+export class TableFacade {
+    private readonly intl: IntlShape;
+
+    public readonly tableDescriptor: TableDescriptor;
+    private readonly resizedColumnsStore: ResizedColumnsStore;
+
+    private currentResult: IExecutionResult;
+    private visibleData: DataViewFacade;
+    private currentFingerprint: string;
+
+    private autoResizedColumns: IResizedColumns;
+    private growToFittedColumns: IResizedColumns;
+    private resizing: boolean;
+    private numberOfColumnResizedCalls: number;
+    private agGridDataSource: AgGridDatasource;
+    private gridApi: GridApi | undefined;
+    private columnApi: ColumnApi | undefined;
+
+    private onPageLoadedCallback: ((dv: DataViewFacade, newResult: boolean) => void) | undefined;
+
+    constructor(
+        result: IExecutionResult,
+        dataView: IDataView,
+        config: TableConfig,
+        props: Readonly<ICorePivotTableProps>,
+    ) {
+        this.intl = props.intl;
```

Comment:
What if props.intl changes? Is there a way to update the instance value?

Reply:
same thing as before: nothing. the intl is used to sanitize empty headers.. which was done once at the initial data processing time.